### ISSUE TITLE
Experimental feature warning

### ISSUE THREAD

**Issue by billc3012** [nodejs/abi-stable-node-addon-examples#12]
When running my N-API code I get this warning

(node:21984) Warning: N-API is an experimental feature and could change at any time.

This is a nuisance.  Can we have an API call to disable the warning message if we dont want it  ?

By the way, I like N-API

**Comment by mhdawson** [comment]
Can you use `--no-warnings` ?  That should suppress the warning along with all others.

I guess you were hoping to be able to disable it for users of your module ?

**Comment by billc3012** [comment]
Thanks

It works as expected  "node --no-warnings index.js"

As you say there might be other warnings I would like to be aware of.

Maybe a parameter in the N-API Init function could suppress the
experimental feature message.

Something like "napi_value Init(napi_env env, napi_value exports,
napi_value experimental)..."  You are better able to judge how to do this
cleanly.

It is early days yet.  I wrote my first N-API code yesterday and it seems
to be what I want to solve my problem.  Getting sensible synchronisation
between a database and a Node application server.  I gave up on examining
how Mongodb did it;  opensource code can be exceedingly complex.

**Comment by Webern** [comment]
I would also like to suppress this warning. It served it's purpose, I understand that I am using an experimental feature, but I am tired of the clutter.  Thanks.

**Comment by mhdawson** [comment]
@Webern my guess is that you are using 8.x, right?

In 6.x it is still doc experimental but there is no longer a warning
In 10.x -> stable no warning
In 8.x -> just waiting for next semverMinor (see https://github.com/nodejs/node/pull/21593).  I've already backported the change to remove the warning, just waiting for the release :)

**Comment by Webern** [comment]
I am using Node v9.8, is that what you mean?

**Comment by mhdawson** [comment]
Since 9.x is no longer supported we'd not be updating it there.  I meant 8.x with 8.12 being the semVer minor release that we are waiting on.

**Comment by Webern** [comment]
Thank you for this info, I resolved the issue by updating my project to 10.7.

**Comment by mhdawson** [comment]
Closing

**Comment by jukefr** [comment]
Is there any way to make this an environment variable of some sorts ?

I am currently writing a unix executable along these lines

```
#!/usr/bin/env node
const fs = require('fs').promises
// more logic...
```

So passing the argument in this case is not an option.

EDIT: https://github.com/nodejs/node/pull/10842 feature was already requested and merged apparently

**Comment by ancms2600** [comment]
In case it was not clear, if you are on the `lts/carbon` `8.x` branch, you can stay on that major version just upgrade the minor version to something `>= 8.12` to resolve this warning.

**Comment by BananaAcid** [comment]
@jukefr another shebang line might help: 
```
#!/bin/sh
':' //; exec "$(command -v node)" "--experimental-modules" "--no-warnings" "$0" "$@"

import fsOrig from 'fs';
const fs = fsOrig.promises;

(async () => {
  let files = await fs.readdir( process.cwd() );
  console.log( files );
})().catch(console.error);
```

I coded [this here](https://gist.github.com/BananaAcid/7ff43ecc26673da3fa64d94b80ab22a4),to test this case.


- `$(command -v node)` finds the executable's full path (do not use `which`), passes it to `exec`, since `exec` can use environment variables (shebang can not), and executes it with all params
- `env` can only take ONE parameter, so this second line is needed.
- `$0` is the script (first param), `$@` all the other manually passed ones from the command line
- works with WSL ;)

**Comment by ackvf** [comment]
I am on node 10.10 and I still get the ExperimentalWarning.

**Comment by mhdawson** [comment]
@ackvf can you open a new issue with the information on how to recreate?  Please include node --version, platform and the command line you used along with what is necessary to recreate. There should be no warning.
